kartothek/core/index.py

```diff
@@ -739,6 +738,28 @@ def filter_indices(index_dict, partitions):
     return new_index_dict
 
 
+def _parquet_bytes_to_dict(column, index_buffer):
+    reader = pa.BufferReader(index_buffer)
+    # This can be done much more efficient but would take a lot more
+    # time to implement so this will be only done on request.
+    table = pq.read_table(reader)
+    column_type = table.schema.field_by_name(column).type
+
+    # `datetime.datetime` objects have a precision of up to microseconds only, so arrow
+    # parses the type to `pa.timestamp("us")`. Since the
+    # values are normalized to `numpy.datetime64[ns]` anyways, we do not care about this
+    # and store the column type as `pa.timestamp("ns")`
+    if column_type == pa.timestamp("us"):
```

Comment:
Not 100% sure if this would fix it but write_table has an option `coerce_timestamp`. Coercing the timestamps to `ns`. Maybe we don't need this check if we are stricter there

http://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html#pyarrow.parquet.write_table

Reply:
`coerce_timestamps` only goes up to `ms` unfortunately